dot/core/service_integration_test.go

```diff
@@ -11,31 +11,147 @@ import (
+	require.NoError(t, err)
+	bKey = append(bKey, h2...)
+	bKey = append(bKey, pub...)
+	return bKey
+}
+
+func newTestDigest(t *testing.T, slotNumber uint64) scale.VaryingDataTypeSlice {
+	testBabeDigest := types.NewBabeDigest()
+	err := testBabeDigest.Set(types.BabePrimaryPreDigest{
+		VRFOutput: [sr25519.VRFOutputLength]byte{
+			0, 91, 50, 25, 214, 94, 119, 36, 71, 216, 33,
+			152, 85, 184, 34, 120, 61, 161, 164, 223, 76,
+			53, 40, 246, 76, 38, 235, 204, 43, 31, 179, 28},
+		VRFProof: [sr25519.VRFProofLength]byte{
+			120, 23, 235, 159, 115, 122, 207, 206, 123, 232,
+			75, 243, 115, 255, 131, 181, 219, 241, 200, 206,
+			21, 22, 238, 16, 68, 49, 86, 99, 76, 139, 39, 0,
+			102, 106, 181, 136, 97, 141, 187, 1, 234, 183, 241,
+			28, 27, 229, 133, 8, 32, 246, 245, 206, 199, 142,
+			134, 124, 226, 217, 95, 30, 176, 246, 5, 3},
+		AuthorityIndex: 17,
+		SlotNumber:     slotNumber,
+	})
+	require.NoError(t, err)
+	data, err := scale.Marshal(testBabeDigest)
+	require.NoError(t, err)
+	vdts := types.NewDigest()
+	err = vdts.Add(
+		types.PreRuntimeDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              data,
+		},
+		types.ConsensusDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              data,
+		},
+		types.SealDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              data,
+		},
+	)
+	require.NoError(t, err)
+	return vdts
+}
+
+func generateTestValidRemarkTxns(t *testing.T, pubKey []byte, accInfo types.AccountInfo) ([]byte, runtime.Instance) {
+	projectRootPath := utils.GetProjectRootPathTest(t) + "/chain/gssmr/genesis.json"
+	gen, err := genesis.NewGenesisFromJSONRaw(projectRootPath)
+	require.NoError(t, err)
+
+	genTrie, err := genesis.NewTrieFromGenesis(gen)
+	require.NoError(t, err)
+
+	genState, err := storage.NewTrieState(genTrie)
+	require.NoError(t, err)
+
+	nodeStorage := runtime.NodeStorage{}
+	nodeStorage.BaseDB = runtime.NewInMemoryDB(t)
+	cfg := &wasmer.Config{
+		InstanceConfig: runtime.InstanceConfig{
+			Storage:     genState,
+			LogLvl:      log.Error,
+			NodeStorage: nodeStorage,
+		},
+		Imports: nil,
+	}
+
+	rt, err := wasmer.NewRuntimeFromGenesis(cfg)
+	require.NoError(t, err)
+
+	aliceBalanceKey := balanceKey(t, pubKey)
+	encBal, err := scale.Marshal(accInfo)
+	require.NoError(t, err)
+
+	rt.(*wasmer.Instance).GetContext().Storage.Set(aliceBalanceKey, encBal)
+	// this key is System.UpgradedToDualRefCount -> set to true since all accounts have been upgraded to v0.9 format
+	rt.(*wasmer.Instance).GetContext().Storage.Set(common.UpgradedToDualRefKey, []byte{1})
+
+	genesisHeader := &types.Header{
+		Number:    big.NewInt(0),
+		StateRoot: genTrie.MustHash(),
+	}
+
+	// Hash of encrypted centrifuge extrinsic
+	extHex := runtime.NewTestExtrinsic(t, rt, genesisHeader.Hash(), genesisHeader.Hash(),
+		0, "System.remark", []byte{0xab, 0xcd})
+
+	extBytes := common.MustHexToBytes(extHex)
+	extBytes = append([]byte{byte(types.TxnExternal)}, extBytes...)
```

Comment:
nit a tiny bit clearer imo

```suggestion
	const txnType = byte(types.TxnExternal)
	extBytes = append([]byte{txnType}, extBytes...)
```